### PR TITLE
fix: updates should include universal2 Pythons

### PR DESCRIPTION
**Pull request by henryiii**
Fixes issues seen in #572

Small cleanup tweaks to black and MyPy, to make running by hand a little easier.

### CODE REVIEW

File: bin/update_pythons.py [hunk 1]
```diff
@@ -229,11 +225,12 @@ def update_config(self, config: Dict[str, str]) -> None:
...
 
         # We need to use ** in update due to MyPy (probably a bug)
-        if "macosx_x86_64" in identifier:
+        if "macos" in identifier:
             if identifier.startswith("pp"):
                 config_update = self.macos_pypy.update_version_macos(spec)
             else:
-                config_update = self.macos_9.update_version_macos(spec) or self.macos_6.update_version_macos(spec)
+                config_update = self.macos_x86_64.update_version_macos(identifier, spec)
+
```
Comment (henryiii) [inline bin/update_pythons.py:233]: This is much faster, especially when adding a check for 11 first.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -21,14 +21,14 @@ repos:
   hooks:
   - id: black
     files: ^bin/update_pythons.py$
-    args: ["--line-length=120"]
```
Comment (YannickJadoul) [inline .pre-commit-config.yaml:24]: Why is this being removed? Feels unrelated.

Comment (YannickJadoul) [reply .pre-commit-config.yaml:24]: Oh, nevermind. Moved down, I see now!

Comment (henryiii) [reply .pre-commit-config.yaml:24]: It had to be manually given if running black directly. Now, it isn't required if running black directly.

File: bin/update_pythons.py [hunk 3]
```diff
@@ -215,9 +213,7 @@ def __init__(self) -> None:
         self.windows_64 = WindowsVersions("64")
         self.windows_pypy = PyPyVersions("32")
 
-        self.macos_6 = CPythonVersions(plat_arch="macosx_x86_64", file_ident="macosx10.6.pkg")
-        self.macos_9 = CPythonVersions(plat_arch="macosx_x86_64", file_ident="macosx10.9.pkg")
-        self.macos_u2 = CPythonVersions(plat_arch="macosx_universal2", file_ident="macos11.0.pkg")
+        self.macos_x86_64 = CPythonVersions()
```
Comment (joerick) [inline bin/update_pythons.py:216]: i guess this variable should be just `self.macos_cpython`? I don't see x86_64 parts...?

**Review by joerick — APPROVED** [review]
Thanks @henryiii - proof is in the runnin' i suppose but this looks okay to me 👍

File: bin/update_pythons.py (hunk 3 above)
Comment (henryiii) [reply bin/update_pythons.py]: Sure.
